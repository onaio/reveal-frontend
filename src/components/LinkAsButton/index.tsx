import * as React from 'react';
import { Link } from 'react-router-dom';
import { NEW_PLAN_URL } from '../../constants';

/** interface for LinkAsButton component props */
export interface Props {
  classNameProp: string;
  text: string;
  to: string;
}

/** LinkAsButton component */
const LinkAsButton = (props: Props) => {
  const { to, classNameProp, text } = props;
  return (
    <Link to={to} className={classNameProp}>
      {text}
    </Link>
  );
};

/** default props for LinkAsButton component */
const defaultProps: Props = {
  classNameProp: 'focus-investigation btn btn-primary float-right mt-5',
  text: 'Add Focus Investigation',
  to: NEW_PLAN_URL,
};
LinkAsButton.defaultProps = defaultProps;

export default LinkAsButton;
