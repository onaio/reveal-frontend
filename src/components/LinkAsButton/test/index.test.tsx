import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Router } from 'react-router';
import LinkAsButton from '..';

const history = createBrowserHistory();

describe('components/LinkToNewPlans', () => {
  it('renders without crashing', () => {
    shallow(<LinkAsButton />);
  });

  it('Matches snapshot', () => {
    const wrapper = mount(
      <Router history={history}>
        <LinkAsButton />
      </Router>
    );
    expect(wrapper.find('Router').length).toEqual(1);
    expect(toJson(wrapper.find('LinkAsButton'))).toMatchSnapshot();
    wrapper.unmount();
  });
});
