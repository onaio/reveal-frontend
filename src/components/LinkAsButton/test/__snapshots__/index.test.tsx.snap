// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/LinkToNewPlans Matches snapshot 1`] = `
<LinkAsButton
  classNameProp="focus-investigation btn btn-primary float-right mt-5"
  text="Add Focus Investigation"
  to="/plans/new"
>
  <Link
    className="focus-investigation btn btn-primary float-right mt-5"
    replace={false}
    to="/plans/new"
  >
    <a
      className="focus-investigation btn btn-primary float-right mt-5"
      href="/plans/new"
      onClick={[Function]}
    >
      Add Focus Investigation
    </a>
  </Link>
</LinkAsButton>
`;
